### ISSUE TITLE
[MySQL] Parsing error when using EVENT statements

### ISSUE THREAD

**Issue by daniellf99** [sqlfluff/sqlfluff#5591]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

The parser was unable to parse EVENT statements from MySQL, such as `CREATE EVENT` or `DROP EVENT`.

### Expected Behaviour

SQLFluff parses and lints the file correctly.

### Observed Behaviour

I created a file that contains the following:

```sql
DROP EVENT test_event;
```
After that, I ran this command:

```shell
sqlfluff lint --dialect mysql ./test.sql -v
```
And the program generated the following message:

```
==== sqlfluff ====
sqlfluff:                2.3.5 python:                 3.8.10
implementation:        cpython verbosity:                   1
dialect:                 mysql templater:               jinja
rules:                                all
==== readout ====

== [test.sql] FAIL                                                              
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'DROP
                       | EVENT'
WARNING: Parsing errors found and dialect is set to 'mysql'. Have you configured your dialect correctly?
==== summary ====
violations:        1 status:         FAIL
```

### How to reproduce

Create a `.sql` file and write the following:

```sql
DROP EVENT test_event;
```
Then run this command:

```shell
sqlfluff lint --dialect mysql ./test.sql -v
```
Replacing `DROP EVENT` with `CREATE EVENT` generates the same error:

```sql
CREATE EVENT e_hourly
    ON SCHEDULE
      EVERY 1 HOUR
    COMMENT 'Clears out sessions table each hour.'
    DO
      DELETE FROM site_activity.sessions;
```

### Dialect

mysql

### Version

sqlfluff, version 2.3.5

Python 3.8.10

### Configuration

None

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by Woodz** [comment]
Also applicable for PSQL variant

**Comment by joaostorrer** [comment]
What do you mean by `PSQL` @Woodz? PostgreSQL?
I did not find `EVENT` statements in PostgreSQL documentation.

**Comment by daniellf99** [comment]
> What do you mean by `PSQL` [@Woodz](https://github.com/Woodz)? PostgreSQL? I did not find `EVENT` statements in PostgreSQL documentation.

In the case @Woodz meant Postgres, there are [CREATE EVENT TRIGGER statements in the docs](https://www.postgresql.org/docs/current/sql-createeventtrigger.html). But that belongs to a separate bug issue in my opinion.

**Comment by joaostorrer** [comment]
@Woodz can you open a separate issue for `CREATE EVENT TRIGGER` statements in Postgres?
